src/network/protocol/messages/block.zig

```diff
@@ -0,0 +1,201 @@
+const std = @import("std");
+const native_endian = @import("builtin").target.cpu.arch.endian();
+const protocol = @import("../lib.zig");
+
+const ServiceFlags = protocol.ServiceFlags;
+
+const read_bytes_exact = @import("../../../util/mem/read.zig").read_bytes_exact;
+
+const Endian = std.builtin.Endian;
+const Sha256 = std.crypto.hash.sha2.Sha256;
+
+const CompactSizeUint = @import("bitcoin-primitives").types.CompatSizeUint;
+const BlockHeader = @import("../../../types/block.zig").BlockHeader;
+const Transaction = @import("../../../types/transaction.zig").Transaction;
+
+/// BlockMessage represents the "block" message
+///
+/// https://developer.bitcoin.org/reference/p2p_networking.html#block
+pub const BlockMessage = struct {
+    block_header: BlockHeader,
+    txn_count: CompactSizeUint,
+    txns: std.ArrayList(Transaction),
```

Comment:
why not a slice?

Reply:
Because when deserializing a block, I am appending to it, and it is easier to append to ArrayList than to a slice.
However, for optimization purpose, I guess I could allocate a slice for the required size, and put my txns inside one by one. Should I do that ?

Reply:
Yes I think it could work.
And you can totally do it in a single call to read

```zig
for (0..bm.txn_count) |i| {
  bm.txns[i] = try std.readStruct(Transaction);
}
```